### PR TITLE
REF: _try_cast; go through fastpath more often

### LINKED ISSUE #28145: Series construction: _try_cast typo, function often taking slow route?
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
from pandas.core.internals.construction import _try_cast

arr = np.arange(0, 10, dtype = "int64")

# should return early
_try_cast(arr, None, False, False)
```
#### Problem description

During series construction, a function, `sanitize_array` attempts to use `_try_cast`, to cast the input to a better type. `_try_cast` is fairly slow to run, so it tries to avoid casting in common cases. However, due to a missing `not` keyword, it appears `_try_cast` **runs for the cases it wants to avoid (like the one above)**.

Here are the relevant lines of `_try_catch`:

https://github.com/pandas-dev/pandas/blob/master/pandas/core/construction.py#L511-L513

Should this be `not maybe_castable(arr)`? 

It is very surprising that lines like this would intentially create an array, and then try to cast it, even when the dtype option passed is None.

https://github.com/pandas-dev/pandas/blob/master/pandas/core/construction.py#L429-L432

#### Expected Output

`_try_cast` not run during `sanitize_array` for common types (e.g. int64). However, from looking at it with `%%prune`, and running pdb, I can see functions like `maybe_cast_to_datetime` are called.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.6.7.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 17.7.0
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.1
numpy            : 1.16.1
pytz             : 2018.9
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 39.0.1
Cython           : None
pytest           : 4.4.2
hypothesis       : None
sphinx           : 2.0.1
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : 0.9.3
psycopg2         : 2.8.2 (dt dec pq3 ext lo64)
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : 3.1.0
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
s3fs             : None
scipy            : 1.2.1
sqlalchemy       : 1.3.4
tables           : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None

</details>

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #28145
- [ ] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
can you post benchmark results

File: pandas/core/construction.py [hunk 1]
```diff
@@ -664,30 +664,45 @@ def _try_cast(
...
+            # TODO: copy?
+
         array_type = dtype.construct_array_type()._from_sequence
         subarr = array_type(arr, dtype=dtype, copy=copy)
         return subarr
 
-    if is_object_dtype(dtype) and not isinstance(arr, np.ndarray):
-        subarr = construct_1d_object_array_from_listlike(arr)
-        return subarr
+    elif is_object_dtype(dtype):
```
Comment (jreback) [inline pandas/core/construction.py:692]: could do this check upfront and assign to a variable (as do this at the top)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/construction.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/construction.py:692]: sure.  might as well optimize to the gills

**Comment by jbrockmendel** [comment]
with `-b ctors`

```
       before           after         ratio
     [6dff995e]       [29848ac6]
     <master>         <ref-cast>
+     2.67±0.05ms       2.74±0.1ms     1.03  ctors.SeriesConstructors.time_series_constructor(<function list_of_tuples at 0x11c088670>, True, 'int')
+     2.66±0.05ms      2.69±0.09ms     1.01  ctors.SeriesConstructors.time_series_constructor(<function list_of_tuples_with_none at 0x11c088820>, True, 'float')
-     2.63±0.05ms      2.60±0.04ms     0.99  ctors.SeriesConstructors.time_series_constructor(<function list_of_lists_with_none at 0x11c0888b0>, True, 'float')
-     2.62±0.05ms      2.59±0.05ms     0.99  ctors.SeriesConstructors.time_series_constructor(<function list_of_lists_with_none at 0x11c0888b0>, False, 'float')
-     2.63±0.03ms      2.59±0.04ms     0.99  ctors.SeriesConstructors.time_series_constructor(<function list_of_lists at 0x11c088790>, False, 'float')
-     2.64±0.05ms      2.60±0.04ms     0.98  ctors.SeriesConstructors.time_series_constructor(<function list_of_lists_with_none at 0x11c0888b0>, False, 'int')
-     2.66±0.07ms      2.61±0.04ms     0.98  ctors.SeriesConstructors.time_series_constructor(<function list_of_lists_with_none at 0x11c0888b0>, True, 'int')
-     2.65±0.07ms      2.59±0.04ms     0.98  ctors.SeriesConstructors.time_series_constructor(<function list_of_lists at 0x11c088790>, True, 'float')
-      3.51±0.1ms      3.38±0.08ms     0.96  ctors.SeriesConstructors.time_series_constructor(<function gen_of_str at 0x11c088550>, False, 'int')
-     2.73±0.08ms      2.62±0.04ms     0.96  ctors.SeriesConstructors.time_series_constructor(<function list_of_tuples_with_none at 0x11c088820>, True, 'int')
-        591±30μs         565±20μs     0.96  ctors.SeriesConstructors.time_series_constructor(<function list_of_str at 0x11c0884c0>, True, 'float')
-      2.77±0.1ms      2.65±0.06ms     0.96  ctors.SeriesConstructors.time_series_constructor(<function list_of_tuples at 0x11c088670>, False, 'int')
-      50.5±0.8μs         45.4±2μs     0.90  ctors.SeriesConstructors.time_series_constructor(<function no_change at 0x11c088430>, True, 'float')
-        43.0±2μs         37.4±2μs     0.87  ctors.SeriesConstructors.time_series_constructor(<function no_change at 0x11c088430>, False, 'float')
-      52.9±0.6μs         44.5±2μs     0.84  ctors.SeriesConstructors.time_series_constructor(<function no_change at 0x11c088430>, True, 'int')
-        45.5±1μs       36.6±0.2μs     0.81  ctors.SeriesConstructors.time_series_constructor(<function no_change at 0x11c088430>, False, 'int')

```

**Review by jreback — APPROVED** [review]

**Pull request merged**